tests/std/tests/P1023R0_constexpr_for_array_comparisons/test.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <array>
+#include <cassert>
+
+using namespace std;
+
+constexpr array<int, 5> a0{2, 8, 9, 1, 9};
+constexpr array<int, 3> a1{2, 8, 9};
+constexpr array<int, 5> a2{2, 8, 9, 1, 8};
+constexpr array<int, 0> a3{};
+
+constexpr bool test_operator_eq() {
+    assert(a0 == a0);
+    assert(a1 == a1);
+    assert(!(a0 == a2));
+    assert(a3 == a3);
+
+    return true;
+}
+
+constexpr bool test_operator_neq() {
+    assert(!(a0 != a0));
+    assert(!(a1 != a1));
+    assert((a0 != a2));
+    assert(!(a3 != a3));
+
+    return true;
+}
+
+constexpr bool test_operator_lt() {
+    assert(!(a0 < a0));
+    assert(!(a1 < a1));
+    assert(a2 < a0);
+    assert(!(a3 < a3));
+
+    return true;
+}
+
+constexpr bool test_operator_gt() {
+    assert(!(a0 > a0));
+    assert(!(a1 > a1));
+    assert(a0 > a2);
+    assert(!(a3 > a3));
+
+    return true;
+}
+
+
+constexpr bool test_operator_leq() {
+    assert(a0 <= a0);
+    assert(a1 <= a1);
+    assert(a2 <= a0);
+    assert(a3 <= a3);
+
+    return true;
+}
+
+
+constexpr bool test_operator_geq() {
+    assert(a0 >= a0);
+    assert(a1 >= a1);
+    assert(a0 >= a2);
+    assert(a3 >= a3);
+
+    return true;
+}
+
+constexpr bool test() {
+    // clang-format off

```

Comment:
Suggestion: I think you could avoid having to disable clang-format, and make your test shorter, by doing the following:
1. Change each of the `test_operator_meow()` functions to return `void` (while still performing their asserts).
2. Change `test()` to call each of the void-returning helpers, then `return true;`.

Reply:
That wont be possible as the return value is required to make it work during both runtime and compile-time

Reply:
I think only the top-level `test()` needs a return value. That is:

```cpp
#include <assert.h>

constexpr void test_is_odd(int x) {
    assert(x % 2 != 0);
}

constexpr bool test() {
    test_is_odd(1729);
    return true;
}

int main() {
    assert(test());
    static_assert(test());
}
```